src/irc.py

```diff
@@ -129,52 +127,121 @@ def _install_snap_package(
+        self, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
+        """Generate the content of the app registration file.
 
-        Returns:
-            A string
+        Args:
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        pass
-
-    def _generate_conf_local(self) -> str:
+        app_reg_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_REGISTRATION_FILE_PATH} ]] || "
+            f"matrix-appservice-irc -r -f {IRC_BRIDGE_REGISTRATION_FILE_PATH}"
+            f" -u https://{matrix.host}:{IRC_BRIDGE_HEALTH_PORT} "
+            f"-c {IRC_BRIDGE_CONFIG_FILE_PATH} -l {config.bot_nickname}",
+        ]
+        logger.info("Creating an app registration file for IRC bridge.")
+        subprocess.run(
+            app_reg_create_command, shell=True, check=True, capture_output=True
+        )  # nosec
+
+    def _eval_conf_local(
+        self, db: DatasourcePostgreSQL, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
         """Generate the content of the irc configuration file.
 
-        Returns:
-            A string
+        Args:
+            db: the database configuration
+            matrix: the matrix configuration
+            config: the charm configuration
+
+        Raises:
+            KeyError: when encountering a KeyError from the configuration file
         """
-        pass
+        with open(f"{IRC_BRIDGE_CONFIG_FILE_PATH}", "r", encoding="utf-8") as config_file:
+            data = yaml.safe_load(config_file)
+        try:
+            db_conn = data["database"]["connectionString"]
+            if db_conn == "" or db_conn != db.uri:
+                db_conn = data["database"]["connectionString"]
+            data["homeserver"]["url"] = f"https://{matrix.host}"
+            data["ircService"]["ident"] = config.ident_enabled
+            data["ircService"]["permissions"] = {}
+            for admin in config.bridge_admins:
+                data["ircService"]["permissions"][admin] = "admin"
+        except KeyError as e:
+            logger.exception("KeyError: {%s}", e)
+            raise
+        with open(f"{IRC_BRIDGE_CONFIG_FILE_PATH}", "w", encoding="utf-8") as config_file:
+            yaml.dump(data, config_file)
 
-    def _generate_app_registration_local(self) -> str:
-        """Generate the content of the app registration file.
+    def reload(self) -> None:
+        """Reload the matrix-appservice-irc service.
 
-        Returns:
-            A string
+        Check if the service is running and reload it.
+
+        Raises:
+            ReloadError: when encountering a SnapError
         """
-        pass
+        try:
+            systemd.service_reload(IRC_BRIDGE_SNAP_NAME)
+        except systemd.SystemdError as e:
+            error_msg = f"An exception occurred when reloading {IRC_BRIDGE_SNAP_NAME}. Reason: {e}"
+            logger.exception(error_msg)
```

Comment:
with logger.exception the exception will be logged, and we raise from e, so I think the reason is not necessary

Reply:
Makes sense, I'll remove it.

Reply:
That pattern is also present in the bind-operator fwiw.